lib/install/management/appliance.go

```diff
@@ -202,6 +205,21 @@ func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine, force bool) error {
 		}
 		log.Infof("Unregistered VM to cleanup after failed destroy: %q", vm.Reference())
 	}
+	if err == nil && d.isVC {
+		// FIXME (caglar10ur): Unregister can throw ConcurrentAccess fault, still debugging but we may end up ignoring it
+		if err := vm.Unregister(d.ctx); err != nil {
+			ok := false
+			if soap.IsSoapFault(err) {
+				// because internal
+				ok = soap.ToSoapFault(err).String == "vim.fault.ConcurrentAccess"
+			}
+			if !ok {
+				return fmt.Errorf("Failed to unregister VM %q: %s", vm.Reference(), err)
+			}
+			log.Debugf("Ignoring ConcurrentAccess error for %s", vm.Reference())
```

Comment:
what gonna happen if concurrent access error is returned? is that vm unregistered or still there